src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -548,4 +549,16 @@ private function registerValidatorConfiguration(ContainerBuilder $container, arr
 
         $container->setParameter('api_platform.validator.serialize_payload_fields', $config['validator']['serialize_payload_fields']);
     }
+
+    /**
+     * Registers the DataCollector configuration.
+     */
+    private function registerDataCollector(ContainerBuilder $container, array $config, XmlFileLoader $loader)
+    {
+        if (!$container->getParameter('kernel.debug')) {
```

Comment:
After reflection, this is the bad way. In fact, `kernel.debug` does not affect the profiler and the data collection. The only way to disable this is with the FrameworkBundle:
```yaml
framework:
  profiler:
    enabled: false
```

So, you can remove this condition but IMO it would be nice to add configuration to enable our data collector (`ArrayNodeDefinition::canBeEnabled()`).

Reply:
@meyerbaptiste done, I added an `enable_profiler` key in the configuration